File: server/models/revenue.go
package models

type Revenue struct {
	ID                uint    `gorm:"primaryKey"`
	UserID            uint    `gorm:"not null"`
	AccountID         uint    `gorm:"not null"`
	Name              string  `gorm:"not null"`
	Description       string  `gorm:"not null"`
	Amount            float32 `gorm:"not null"`
	DepositDate       int     `gorm:"not null"`
	Recurring         bool    `gorm:"not null"`
	RecurringInterval int     `gorm:"not null"`
	CreatedAt         int
	UpdatedAt         int
}

File: entity/src/revenue.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Deserialize, Serialize)]
#[sea_orm(table_name = "revenue")]
pub struct Model {
    #[sea_orm(primary_key)]
    #[serde(skip_deserializing)]
    pub id: i32,
    pub user_id: Option<i32>,
    pub account_id: Option<i32>,
    pub name: String,
    pub description: String,
    // Update to float
    pub amount: i64,
    pub deposit_date: i64,
    pub is_recurring: bool,
    pub recurring_interval: i64,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(
        belongs_to = "super::user::Entity",
        from = "Column::UserId",
        to = "super::user::Column::Id",
    )]
    User,
    #[sea_orm(
        belongs_to = "super::account::Entity",
        from = "Column::AccountId",
        to = "super::account::Column::Id"
    )]
    Account,
}

impl Related<super::user::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::User.def()
    }
}

impl Related<super::account::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::Account.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}
